protocol/signalfx/format/log/signalfx_log.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package sfx_log_format;
+
+option go_package = "gen_go/;log_format";
+
+// LogRequest represents the data that is incoming which is a list of ResourceLogs
+message LogRequest {
+    repeated ResourceLogs logs = 1;
+}
+
+// ResourceLogs handles either sharing a resource with list of LogRecord or unique for each logRecord
+message ResourceLogs {
+    Resource resource = 1;
+    repeated LogRecord logs = 2;
+}
+
+// model is based off https://github.com/open-telemetry/oteps/blob/master/text/0097-log-data-model.md#log-and-event-record-definition
+message LogRecord {
+    fixed64 Timestamp = 1;
+    bytes TraceID = 2;
+    bytes SpanID = 3;
+    fixed32 TraceFlags = 4;
+    string SeverityText = 5;
+    fixed32 SeverityNumber = 6;
+    string Name = 7;
+    Body Body = 8;
+    Attributes Attributes = 9;
+}
+
+// AttributeValue is used to store Body attribute different types of values
+// @TODO: benchmark protobuf auto-gen with gogo/protobuf
+message AttributeValue {
+    // ValueType is the enumeration of possible types that value can have.
+    enum ValueType {
+        STRING = 0;
+        INT    = 1;
+        DOUBLE = 2;
+        LIST   = 3;
+        MAP    = 4;
+        BOOL   = 5;
+    }
+
+    // type of the value
+    ValueType type = 1;
+    // Only one of the following fields is supposed to contain data (determined by `type` field).
+    // This is deliberately not using Protobuf `oneof`
+
+    string v_string = 2;
+    int64 v_long = 3;
+    double v_double = 4;
+    bool v_bool = 5;
+    AttributeList v_list = 6;
+    AttributeMap v_map = 7;
+}
+
+message Body {
+    AttributeValue values = 1;
+}
+
+message Resource {
+    map<string, AttributeValue> values = 1;
+}
+
+message Attributes {
+    map<string, AttributeValue> values = 1;
```

Comment:
Maps are the slowest data type in Go's ProtoBuf implementation . I recommend to use a `repeated` message that contains a key-value pair unless you strongly care about key uniqueness on the wire (you probably don't).
From the top of my head we gained about 3x _overall_ serialization/deserialization speed improvement just from this single change in the past.
Again, don't trust my word, benchmark it :-)

Reply:
the only downside is lookups with `repeated` is looking up a Attribute will be iterating over the list.

Reply:
for now I will make it as a `repeated` with {key, value} and if we need to do lookups we can always modify this later on.